src/main/java/net/sf/jabref/logic/integrity/IntegrityCheck.java

```diff
@@ -321,6 +324,12 @@ private BracketChecker(String field) {
                 return Collections.singletonList(new IntegrityMessage(Localization.lang("should contain a four digit number"), entry, FieldName.YEAR));
             }
 
+            if (!ENDS_WITH_FOUR_DIGIT.test(value.get().replaceAll(regex, ""))) {
```

Comment:
Something bothered me with the replaceall and the list of regex chars...
After thinking about it for a while I found why:
The first argument of replaceAll is a regex, so your regex char string would be interpreted as actual pattern.
To prevent this, you have to escape the chars, or better use http://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#quote-java.lang.String-